benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -444,42 +471,55 @@ void GraphicsBenchmarkApp::SetupSpheresPipelines()
-    gpCreateInfo.PS                                 = {psQuads.Get(), "psmain"};
-    gpCreateInfo.vertexInputState.bindingCount      = 1;
-    gpCreateInfo.vertexInputState.bindings[0]       = mFullscreenQuads.vertexBinding;
-    gpCreateInfo.topology                           = grfx::PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP;
-    gpCreateInfo.polygonMode                        = grfx::POLYGON_MODE_FILL;
-    gpCreateInfo.cullMode                           = grfx::CULL_MODE_BACK;
-    gpCreateInfo.frontFace                          = grfx::FRONT_FACE_CW;
-    gpCreateInfo.depthReadEnable                    = true;
-    gpCreateInfo.depthWriteEnable                   = false;
-    gpCreateInfo.blendModes[0]                      = grfx::BLEND_MODE_NONE;
-    gpCreateInfo.outputState.renderTargetCount      = 1;
-    gpCreateInfo.outputState.renderTargetFormats[0] = GetSwapchain()->GetColorFormat();
-    gpCreateInfo.outputState.depthStencilFormat     = GetSwapchain()->GetDepthFormat();
-    gpCreateInfo.pPipelineInterface                 = mFullscreenQuads.pipelineInterface;
-    PPX_CHECKED_CALL(GetDevice()->CreateGraphicsPipeline(&gpCreateInfo, &mFullscreenQuads.pipeline));
+        PPX_CHECKED_CALL(GetDevice()->CreatePipelineInterface(&piCreateInfo, &mQuadsPipelineInterfaces[1]));
+    }
+    // Texture
+    {
+        grfx::PipelineInterfaceCreateInfo piCreateInfo = {};
+        piCreateInfo.setCount                          = 1;
+        piCreateInfo.sets[0].set                       = 0;
+        piCreateInfo.sets[0].pLayout                   = mFullscreenQuads.descriptorSetLayout;
+
+        PPX_CHECKED_CALL(GetDevice()->CreatePipelineInterface(&piCreateInfo, &mQuadsPipelineInterfaces[2]));
+    }
+
+    for (size_t i = 0; i < kFullscreenQuadsType.size(); i++) {
+        grfx::GraphicsPipelineCreateInfo2 gpCreateInfo  = {};
+        gpCreateInfo.VS                                 = {mVSQuads.Get(), "vsmain"};
+        gpCreateInfo.PS                                 = {mQuadsPs[i].Get(), "psmain"};
+        gpCreateInfo.vertexInputState.bindingCount      = 1;
+        gpCreateInfo.vertexInputState.bindings[0]       = mFullscreenQuads.vertexBinding;
+        gpCreateInfo.topology                           = grfx::PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP;
+        gpCreateInfo.polygonMode                        = grfx::POLYGON_MODE_FILL;
+        gpCreateInfo.cullMode                           = grfx::CULL_MODE_BACK;
+        gpCreateInfo.frontFace                          = grfx::FRONT_FACE_CW;
+        gpCreateInfo.depthReadEnable                    = true;
+        gpCreateInfo.depthWriteEnable                   = false;
+        gpCreateInfo.blendModes[0]                      = grfx::BLEND_MODE_NONE;
+        gpCreateInfo.outputState.renderTargetCount      = 1;
+        gpCreateInfo.outputState.renderTargetFormats[0] = GetSwapchain()->GetColorFormat();
+        gpCreateInfo.outputState.depthStencilFormat     = GetSwapchain()->GetDepthFormat();
```

Comment:
in this case we probably dont need the depth attachment

Reply:
If I remove depth stuff I actually get different behaviour for the single renderpass setting.

There are a bunch of draw calls of the same plane at the same depth. Without depth testing there is no order to which shows up on top and I've noticed that the color stays the same. Probably the first quad drawn always shows at the front, which I assume isn't the behaviour we want for these quads.

Reply:
hmm, that is weird. without depth testing, the order should be following the drawing order, so the last drawcall should be shown. you can verify this with renderdoc capture to see if this is the case.

Reply:
I tried this and I looked at pixel history, it is only touched by the draw call for the first quad.

Reply:
the multipass one is wrong, we should not do clear when we begin a new pass, should be also `ATTACHMENT_LOAD_OP_DONT_CARE` for each renderpass, and we don't need the depth attachment

Reply:
OK, after our discussion I played around with this a little more and figured out that disabling depth read is fine but changing this depth stencil format will cause weird behaviour on my machine. So I am thinking of removing the depth read but leaving the format alone for now. This is how the other BigWheels benchmarks related to texture loading seem to be handling it (`texture_load` and `texture_sample`)

Reply:
yeah, i checked the code, with the bigwheels framework, we can only disable the depth attachment for the swapchain which will also disable the depth for the sphere pass which is not we want. So for now it is fine by just setting the depth read to false.